### PR TITLE
add rbac perms for IngressClass resource

### PR DESCRIPTION
**Pull request by acookin**
## Description
Fixes https://github.com/datawire/ambassador/issues/3140 along with  https://github.com/datawire/ambassador-chart/pull/157

I think this happened because there seem to be several sources of truth for rbac. This was only added to the ambasssador-rbac initially, and not the aes or the ambassador chart https://github.com/datawire/ambassador/pull/2406/files :(

We could probably have the helm chart be the single source of truth and generate the flat yaml from that? At least for AES? I haven't really looked into what the diff is of those two sources, but it seems to me like this is bound to happen again if we do nothing.

## Related Issues
https://github.com/datawire/ambassador/issues/3140

## Testing
Deployed aes on a 1.19 k8s cluster from the yaml resources on master, and you see ambassador fail to go ready with the logs littered with messages like this:
```
E0108 18:20:17.438700       1 reflector.go:178] k8s.io/client-go@v0.18.4/tools/cache/reflector.go:125: Failed to list *unstructured.Unstructured: ingressclasses.networking.k8s.io is forbidden: User "system:serviceaccount:ambassador:ambassador" cannot list resource "ingressclasses" in API group "networking.k8s.io" at the cluster scope
```
Then after deploying the changes from this branch, the errors go away and I see ambassador go ready:
```
NAME                          READY   STATUS    RESTARTS   AGE
ambassador-588f9ff95d-k6jww   1/1     Running   0          2m11s
```

## Tasks That Must Be Done
- [x] Did you update CHANGELOG.md?
  + [ ] Is this a new feature?
  + [ ] Are there any non-backward-compatible changes?
  + [ ] Did the envoy version change?
  + [ ] Are there any deprecations?
  + [ ] Will the changes impact how ambassador performs at scale?
    - [ ] Might an existing production deployment need to change:
      + memory limits?
      + cpu limits?
      + replica counts?
    - [ ] Does the change impact data-plane latency/scalability?
- [x] Is your change adequately tested?
  + [ ] Was their sufficient test coverage for the area changed?
  + [ ] Do the existing tests capture the requirements for the area changed?
  + [ ] Is the bulk of your code covered by unit tests?
  + [ ] Does at least one end-to-end test cover the integration points your change depends on?
- [x] Did you update documentation?
- [x] Were there any special dev tricks you had to use to work on this code efficiently?
  + [x] Did you add them to DEVELOPING.md?
- [ ] Is this a build change?
  + [ ] If so, did you test on both Mac and Linux?

## Other
* If this is a documentation change for a particular release, please open the pull request against the release branch.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ Please see the [Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest
 
 ## Next Release
 
-(no changes yet)
+- Bugfix: Ambassador now has appropriate permissions on IngressClass resources.
```
Comment (kflynn) [inline CHANGELOG.md:73]: Weirdly, this isn't actually necessary -- the YAML will get published live on merge.

**Review by kflynn — APPROVED** [review]
We don't actually need the `CHANGELOG` update, but the content looks fine, thanks!! 🙂

File: CHANGELOG.md (hunk 1 above)
Comment (acookin) [reply CHANGELOG.md:73]: oh shoot, right. this was muscle memory mostly heh

**Review by kflynn — APPROVED** [review]
Hit it!

**Comment by acookin** [comment]
nope this isn't right. these changes will get blown away when someone runs make update-yaml :(

**Pull request closed without merging**